src/app/Header.js

```diff
@@ -59,17 +60,29 @@ function NavLink({link}) {
 
 function LogOutLink() {
   const theme = useTheme()
+  const router = useRouter();
   const { enqueueSnackbar } = useSnackbar();
   const { setAuthToken } = useContext(AuthTokenContext);
+
```

Comment:
It does not cover a case, when I am trying to access some page with an invalid token, it covers only logout link. It shourd redirect all 401 requests to login.